src/Ipac_Table_Writer.hxx

```diff
@@ -42,38 +63,84 @@ public:
     static std::vector<size_t> get_column_widths(const Table &table);
 
 
+    /*******************************************/
+    /* Helper class */
+    /*******************************************/
+
+    // This class produces the appropriate list of columns to be included in a subtable
+    // (either column_ids itself or its complement, depending on whether
+    // is_col_inclusion_list is true or false).
+
+    class Column_List_Handler {
+    public:
+        Column_List_Handler(const std::set<size_t> &column_ids,
+                            bool is_col_inclusion_list, size_t num_table_columns);
+
+        const std::set<size_t> &get_included_columns_list() {
+            if (is_col_inclusion_list_) {
+                return specified_column_ids_;
+            }
+            return complementary_column_ids_;
+        }
+
+        static std::set<size_t> get_all_column_ids(size_t num_table_columns) {
+            size_t all_column_ids[num_table_columns - 1];
```

Comment:
I think there is a GNU extension for variable length arrays, but I don't think they are standard C++ (my guess is your code will not compile with `-std=c++14 -pedantic`). Also, if `num_table_columns` is `<= 1`, you are invoking undefined behavior (at least for C's version of VLAs).

Please switch to `std::vector`.